client/common/js/articleScroller.js

```diff
@@ -0,0 +1,93 @@
+class ArticleScroller {
+	constructor() {
+		// Since JS is loaded we can show the interactive horizontal scroller
+		document.querySelector('.blog-list__posts--featured').classList.add('blog-list__posts--featured-flex')
+		document.querySelector('.blog-list__featured-scroller').classList.add('blog-list__featured-scroller--visible')
+
+		this.scrollNextBtn = document.querySelector('.blog-list__featured-scroller--button-next')
+		this.scrollPrevBtn = document.querySelector('.blog-list__featured-scroller--button-prev')
+
+		this.articleParent = document.querySelector('.blog-list__posts--featured-flex')
+		this.articles = this.articleParent.querySelectorAll('.blog-list__item--featured')
+		this.perArticleWidth = this.articles[0].getBoundingClientRect().width
+		this.shift = 0
+
+		this.scrollToLeft = this.scrollToLeft.bind(this)
+		this.scrollToRight = this.scrollToRight.bind(this)
+
+		if (!this.prevExists()) {
+			this.scrollPrevBtn.disabled = true
+		}
+
+		if (!this.nextExists()) {
+			this.scrollNextBtn.disabled = true
+		}
+
+		this.scrollNextBtn.addEventListener('click', this.scrollToLeft)
+		this.scrollPrevBtn.addEventListener('click', this.scrollToRight)
+	}
+
+	static isVisible(article) {
+		const rect = article.getBoundingClientRect()
+		if (rect.x < 0 || rect.right > window.innerWidth) {
+			return false
+		}
+		return true
+	}
+
+	shiftArticles(shift) {
+		this.articleParent.style.transform = 'translateX(' + shift + ')'
+	}
+
+	prevExists() {
+		return !ArticleScroller.isVisible(this.articles[0])
+	}
+
+	nextExists() {
+		return !ArticleScroller.isVisible(this.articles[this.articles.length - 1])
+	}
```

Comment:
So I'm not sure if this is worth handling right now, but I did notice that these functions aren't run again if the window is resized, so if the window changes from where the first and/or last article is visible to where it's not visible, the corresponding button will remain inactive and the person won't be able to scroll. This might be an edge case, though, but I just wanted to make a note of that here.

Reply:
Yes. I kind of didn't handle it because of the edge case. But since we don't expect a lot of users to keep resizing their window, I didn't put too much time in that. Let me know if it's necessary to be put in this PR for merging, or you think we can tackle this in a separate PR. Also, I know @harrislapiroff said that he had a similar implementation needed for homepage for horizontal scrolling.

Reply:
I don't think it's needed for this PR, I just didn't want to forget about it

Reply:
maybe we can create an issue and capture this before merging.